x-pack/plugin/gpu/src/main/java/org/elasticsearch/xpack/gpu/GPUPlugin.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.gpu;
+
+import org.apache.lucene.codecs.KnnVectorsFormat;
+import org.apache.lucene.util.hnsw.HnswGraphBuilder;
+import org.elasticsearch.common.util.FeatureFlag;
+import org.elasticsearch.index.IndexSettings;
+import org.elasticsearch.index.mapper.vectors.DenseVectorFieldMapper;
+import org.elasticsearch.index.mapper.vectors.VectorsFormatProvider;
+import org.elasticsearch.plugins.MapperPlugin;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.xpack.gpu.codec.ES92GpuHnswSQVectorsFormat;
+import org.elasticsearch.xpack.gpu.codec.ES92GpuHnswVectorsFormat;
+
+public class GPUPlugin extends Plugin implements MapperPlugin {
```

Comment:
I would rather us have an "InternalVectorFormatProviderPlugin" type thing that only provides this required interface and only this plugin is allowed to extend.